openedx/core/djangoapps/auth_exchange/tests/test_views.py

```diff
@@ -180,3 +185,53 @@ def test_dot_password_grant_supported(self):
     def test_dot_client_credentials_unsupported(self):
         access_token = self._create_dot_access_token()
         self._verify_response(access_token, expected_status_code=401)
+
+    def _create_jwt_token(self, grant_type='password', scope='email profile', use_asymmetric_key=True):
+        """
+        Create jwt token
+        """
+        access_token = self._create_dot_access_token(grant_type)
+        oauth_adapter = DOTAdapter()
+        token_dict = {
+            'access_token': access_token,
+            'scope': scope,
+        }
+        jwt_token = jwt_api.create_jwt_from_token(token_dict, oauth_adapter, use_asymmetric_key=use_asymmetric_key)
+        return jwt_token
+
+    def test_invalid_jwt(self):
+        self._verify_response("invalid_token", token_type="JWT", expected_status_code=401)
+
+        assert 'session_key' not in self.client.session
+
+    def test_valid_jwt_with_not_password_grant(self):
+
+        jwt_token = self._create_jwt_token(grant_type='Client credentials')
+        self._verify_response(jwt_token, token_type="JWT", expected_status_code=401)
+
+        assert 'session_key' not in self.client.session
+
+    def test_expired_jwt_with_password_grant(self):
+        with override_settings(JWT_ACCESS_TOKEN_EXPIRE_SECONDS=1):
+            jwt_token = self._create_jwt_token()
+
+        # freeze_time will pretend 10 seconds have passed!
+        with freeze_time(lambda: datetime.utcnow() + timedelta(seconds=10)):
+            self._verify_response(jwt_token, token_type="JWT", expected_status_code=401)
+
+        assert 'session_key' not in self.client.session
+
+    def test_valid_jwt_with_password_grant(self):
```

Comment:
* Consider renaming test to `test_success_with_valid_jwt_with_password_grant` and other tests with prefix `test_failure_with...", like `test_failure_with_valid_symmetric_jwt`.
* Consider moving this test to the end so all the failure cases are together.